plugins/by-name/otter/default.nix

```
@@ -101,10 +101,16 @@ helpers.neovim-plugin.mkNeovimPlugin {
 
   extraConfig = cfg: {
     warnings =
-      lib.optional (cfg.enable && config.plugins.treesitter.settings.highlight.enable == null)
+      lib.optional
+        (
+          cfg.enable
+          && (
+            !config.plugins.treesitter.enable || config.plugins.treesitter.settings.highlight.enable == null
```

```suggestion
            config.plugins.treesitter.enable -> config.plugins.treesitter.settings.highlight.enable == null
```

`!x || y` is the same as `x -> y`

---

Would need to double check [operator precedence](https://nix.dev/manual/nix/2.17/language/operators) though, as these mean different things:

```nix
(x -> y) ==  null
x -> (y == null)
```
One is the same as `x -> y == null`, but which one?

---

Ok, equality has prio 11 while logical `||` and `->` have 13 and 14 respectively. Therefore parens are not needed.

`x -> (y == null)` and `x -> y == null` are equivalent. Which makes intuitive sense, too.

---

Thanks, that looks cleaner. :)